02_bubble/joel00wood/main_test.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+)
+
+// enter borrowed code
+// https://gist.github.com/hauxe/e935a7f9012bf2649710cf75af323dbf
+// stdOut capture helper
+func captureOutput(f func()) string {
```

Comment:
this is an overkill.
you can 
```
var out io.Writer = os.Stdout
fmt.Fprintln(out, "my lovely string")
```

and 
```
var buf bytes.Buffer
out = &buf
...
printedString := strconv.Quote(buf.String())
```

Reply:
I've implemented this in all of my labs if not solely for the purpose of meeting the requirements of `main()` must `fmt.Println` as opposed to `fmt.Fprintln` which would otherwise enable the output capture as you described. I don't disagree that the overall implementation is overkill in that it covers a lot more cases than is entirely necessary but for the purpose of capturing the output of a given function for testing it fits the bill perfectly.

That and I'd further add that I kinda missed the example you've described as listed in the hello world app provided in the repo so I felt this was the best way of accurately testing the functions as described.